custom_components/meural/pymeural.py

```
@@ -43,19 +51,31 @@ async def request(self, method, path, data=None) -> Dict:
             else:
                 kwargs["json"] = data
         with async_timeout.timeout(10):
-            resp = await self.session.request(
-                method,
-                url,
-                headers={
-                    "Authorization": f"Token {self.token}",
-                    "x-meural-api-version": "3",
-                },
-                raise_for_status=True,
-                **kwargs,
-            )
+            try:
+                resp = await self.session.request(
+                    method,
+                    url,
+                    headers={
+                        "Authorization": f"Token {self.token}",
+                        "x-meural-api-version": "3",
+                    },
+                    raise_for_status=True,
+                    **kwargs,
+                )
+            except ClientResponseError as err:
+                _LOGGER.warning('Meural: Sending Request failed. Re-Authenticating.')
+                self.token = None
+                return
+            except Exception as err:
```

We shouldn't swallow errors and then return bad data. The caller won't know what happened. We should create our own exception that the device is turned off and then raise that.

Also, `Exception` is too generic. Match the exact exception that's thrown when your meural is off and convert it to the DeviceTurnedOff exception

---

done